mtp_api/apps/transaction/api/bank_admin/serializers.py

```diff
@@ -50,3 +51,75 @@ class Meta:
             'reference',
             'received_at'
         )
+
+
+class UpdateRefundedTransactionListSerializer(serializers.ListSerializer):
+
+    @transaction.atomic
+    def update(self, instance, validated_data):
+        user = self.context['request'].user
+
+        to_refund = [t['id'] for t in validated_data if t['refunded']]
+
+        update_set = Transaction.objects.filter(
+            pk__in=to_refund,
+            **Transaction.STATUS_LOOKUP['refund_pending']).select_for_update()
+        if len(update_set) != len(to_refund):
+            raise Transaction.DoesNotExist(
+                list(set(to_refund) - {t.id for t in update_set})
+            )
+
+        update_set.update(refunded=True)
+        for t_id in to_refund:
+            transaction = Transaction()
```

Comment:
I think this might be dangerous as the signal handler wouldn't know that the transaction object only has the ID set.